src/services/associationsService.ts

```diff
@@ -0,0 +1,121 @@
+import { Request } from "express";
+import { createOauthPrivateApiClient } from "./apiClientService";
+import { Resource } from "@companieshouse/api-sdk-node";
+import logger from "../lib/Logger";
+import { StatusCodes } from "http-status-codes";
+import * as constants from "../constants";
+import { Associations, AssociationStatus, Errors, NewAssociationResponse } from "@companieshouse/private-api-sdk-node/dist/services/associations/types";
+
+export const getUserAssociations = async (req: Request, status: AssociationStatus[]): Promise<Associations> => {
+    const apiClient = createOauthPrivateApiClient(req);
+
+    logger.info(`Looking for associations with status ${JSON.stringify(status)}`);
+    const sdkResponse: Resource<Associations | Errors> = await apiClient.associationsService.searchAssociations(status);
+
+    if (!sdkResponse) {
+        logger.error(`Associations API for status ${JSON.stringify(status)}`);
+        return Promise.reject(sdkResponse);
+    }
+
+    if (sdkResponse.httpStatusCode !== StatusCodes.OK) {
+        logger.error(`Http status code ${sdkResponse.httpStatusCode} - Failed to get associations with status ${JSON.stringify(status)}`);
+        return Promise.reject(sdkResponse);
+    }
+
+    if (!sdkResponse.resource) {
+        logger.error(`Associations API returned no resource for associations with status ${JSON.stringify(status)}`);
+        return Promise.reject(sdkResponse);
+    }
+
+    logger.debug(`Received associations ${JSON.stringify(sdkResponse)}`);
+
+    return Promise.resolve(sdkResponse.resource as Associations);
+};
+
+export const isCompanyAssociatedWithUser = async (req: Request, companyNumber: string): Promise<string> => {
```

Comment:
is this a get /associations or get /associations/{company_number}. if latter then you can pass user email as parameer.

Reply:
I modified this function to use /associations/companies/{company_number}?user_email= endpoint